### PR TITLE
Reviews  - Review tile features 

### PR DESCRIPTION
**Pull request by Pallavi25Kishore**
@mictro97 @shaoyuchen3213 
Pull request (commits 1 to 4) cover following functionalities - 
1. Have set up environment to run jest tests (please run npm install after pulling these changes in your local branch)
2. Have completed implementing functionality for displaying stars in the review tiles as per rating. 
3. Have added test suite for Star Rating component.

@mictro97 @shaoyuchen3213 
Adding commits 5, 6 and 7 to the pending pull request for

1.Implementing createdAt date format feature and adding corresponding test
2. Implementing review summary feature and adding corresponding test

@mictro97 @shaoyuchen3213 
Adding commits 8,9, 10 and 11 to pending pull request for - 
1. Adding a different icon for fully filled star 
2. Adding review body text functionality to review tile 
3. Adding review body photo thumbnail functionality to review tile 
4. Adding photo modal window functionality to review tile 
5. adding corresponding tests 

Note: All 13 tests added for the above commits (commit no. 1 to 11) are passing.

### CODE REVIEW

**Comment by Pallavi25Kishore** [comment]
@shaoyuchen3213 @mictro97 
Adding commits 5, 6 and 7 to the pending pull request for 
1. Implementing createdAt date format feature and adding corresponding test
2. Implementing review summary feature and adding corresponding test

File: client/src/components/Reviews/Modal.jsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+const Modal = ({photo, handleCrossClick}) => {
+
+  return (
+    <div className="modalOverlay">
+      <div className="modalContainer" data-testid="modal-container">
+        <div className="crossButtonContainer">
+          <button aria-label="x" onClick={handleCrossClick} className="crossToClose">&#10005;</button>
+        </div>
+        <div className="imageContainer">
+          <img className="modalImage" src={photo.url}></img>
```
Comment (mictro97) [inline client/src/components/Reviews/Modal.jsx:12]: I would consider adding alt text to the image tag if the image can't be displayed

File: client/src/components/Reviews/ReviewBody.jsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    event.preventDefault();
+    setShowMore(true);
+  };
+
+  const handleShowLessClick = (event) => {
+    event.preventDefault();
+    setShowMore(false);
+  };
+
+  const isValidUrl = (url) => {
```
Comment (mictro97) [inline client/src/components/Reviews/ReviewBody.jsx:18]: Could put this function outside the component so it doesn't have to be recreated every render

File: client/src/components/Reviews/ReviewTile.jsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import React from "react";
+import StarRating from "./StarRating.jsx";
+import CreatedAt from "./CreatedAt.jsx";
+import ReviewSummary from "./ReviewSummary.jsx";
+import ReviewBody from "./ReviewBody.jsx";
+
+const ReviewTile = ({review}) => {
+  return (
+    <>
```
Comment (mictro97) [inline client/src/components/Reviews/ReviewTile.jsx:9]: Maybe add a div here with a className to make styling easier

**Review by mictro97 — CHANGES_REQUESTED** [review]
Everything looks great! As nitpicks I would just change these few small things.

File: client/src/components/Reviews/ReviewTile.jsx (hunk 3 above)
Comment (Pallavi25Kishore) [reply client/src/components/Reviews/ReviewTile.jsx]: done, thanks for the suggestion :)

File: client/src/components/Reviews/Modal.jsx (hunk 1 above)
Comment (Pallavi25Kishore) [reply client/src/components/Reviews/Modal.jsx]: done :)

File: client/src/components/Reviews/ReviewBody.jsx (hunk 2 above)
Comment (Pallavi25Kishore) [reply client/src/components/Reviews/ReviewBody.jsx]: sounds good, done :)

**Review by mictro97 — APPROVED** [review]
Looks good!

**Pull request merged**